hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -1300,4 +1171,126 @@ private String getStateMessage() {
                 + " loadedOnCreate=" + loadedOnCreate + " loadedOnPreMigration=" + loadedOnPreMigration
                 + " isLoaded=" + isLoaded();
     }
+
+    @Override
+    public int clear() {
+        checkIfLoaded();
+        // we don't remove locked keys. These are clearable records.
+        Collection<Record> clearableRecords = getNotLockedRecords();
+        // This conversion is required by mapDataStore#removeAll call.
+        List<Data> keys = getKeysFromRecords(clearableRecords);
+        mapDataStore.removeAll(keys);
+        clearMapStore();
+        removeIndex(clearableRecords);
+        return removeRecords(clearableRecords);
+    }
+
+    @Override
+    public void reset() {
+        clearMapStore();
+        storage.clear(false);
+        stats.reset();
+        mutationObserver.onReset();
+    }
+
+    @Override
+    public void destroy() {
+        clearPartition(false, true);
+        storage.destroy(false);
+        mutationObserver.onDestroy(false);
+    }
+
+    @Override
+    public void destroyInternals() {
+        clearMapStore();
+        clearStorage(false);
+        storage.destroy(false);
+        mutationObserver.onDestroy(true);
+    }
+
+    @Override
+    public void clearPartition(boolean onShutdown, boolean onRecordStoreDestroy) {
+        clearOtherResourcesThanStorage(onRecordStoreDestroy);
+        clearStorage(onShutdown || onRecordStoreDestroy);
+    }
+
+    public void clearOtherResourcesThanStorage(boolean onRecordStoreDestroy) {
+        clearMapStore();
+        clearLockStore();
+        clearIndexedData(onRecordStoreDestroy);
```

Comment:
It seems `clearIndexedData` is called unconditionally, while in the original code it was done only in the `onRecordStoreDestroy=false` case. Is it safe to do this way?

Reply:
Ah, I see that the condition has just moved into the method.